radio/src/targets/taranis/hal.h

```diff
@@ -2036,131 +2003,114 @@
-  #define INTMODULE_HEARTBEAT_EXTI_PinSource      GPIO_PinSource1
-  #define INTMODULE_HEARTBEAT_EXTI_LINE           EXTI_Line1
-  #define INTMODULE_HEARTBEAT_EXTI_IRQn           EXTI1_IRQn
-  #define INTMODULE_HEARTBEAT_EXTI_IRQHandler     EXTI1_IRQHandler
+  #define INTMODULE_HEARTBEAT_GPIO_PIN            LL_GPIO_PIN_1  // PB.01
+  #define INTMODULE_HEARTBEAT_EXTI_PORT           LL_SYSCFG_EXTI_PORTB
+  #define INTMODULE_HEARTBEAT_EXTI_SYS_LINE       LL_SYSCFG_EXTI_LINE1
+  #define INTMODULE_HEARTBEAT_EXTI_LINE           LL_EXTI_LINE_1
+  // INTMODULE_HEARTBEAT_EXTI IRQ
+  #if !defined(USE_EXTI1_IRQ)
+    #define USE_EXTI1_IRQ
+    #define EXTI1_IRQ_Priority 5
+  #endif
 #elif defined(RADIO_ZORRO) || defined(RADIO_TX12MK2)
   #define INTMODULE_HEARTBEAT_RCC_AHB1Periph      0
 #else
   #define INTMODULE_HEARTBEAT
   #define INTMODULE_HEARTBEAT_RCC_AHB1Periph      RCC_AHB1Periph_GPIOC
   #define INTMODULE_HEARTBEAT_GPIO                GPIOC
-  #define INTMODULE_HEARTBEAT_GPIO_PIN            GPIO_Pin_7  // PC.07
-  #define INTMODULE_HEARTBEAT_EXTI_PortSource     EXTI_PortSourceGPIOC
-  #define INTMODULE_HEARTBEAT_EXTI_PinSource      GPIO_PinSource7
-  #define INTMODULE_HEARTBEAT_EXTI_LINE           EXTI_Line7
-  #define INTMODULE_HEARTBEAT_EXTI_IRQn           EXTI9_5_IRQn
-  #if defined(RADIO_X7) || defined(RADIO_TX12) || defined(RADIO_TPRO)
-    #define INTMODULE_HEARTBEAT_REUSE_INTERRUPT_ROTARY_ENCODER
-  #else
-    #define INTMODULE_HEARTBEAT_EXTI_IRQHandler     EXTI9_5_IRQHandler
+  #define INTMODULE_HEARTBEAT_GPIO_PIN            LL_GPIO_PIN_7  // PC.07
+  #define INTMODULE_HEARTBEAT_EXTI_PORT           LL_SYSCFG_EXTI_PORTC
+  #define INTMODULE_HEARTBEAT_EXTI_SYS_LINE       LL_SYSCFG_EXTI_LINE7
+  #define INTMODULE_HEARTBEAT_EXTI_LINE           LL_EXTI_LINE_7
+  // INTMODULE_HEARTBEAT_EXTI IRQ
+  #if !defined(USE_EXTI9_5_IRQ)
+    #define USE_EXTI9_5_IRQ
+    #define EXTI9_5_IRQ_Priority 5
   #endif
 #endif
 
-#if defined(PCBX9LITE)
-  #define EXTMODULE_HEARTBEAT_RCC_AHB1Periph      RCC_AHB1Periph_GPIOD
-  #define EXTMODULE_HEARTBEAT_GPIO                GPIOD
-  #define EXTMODULE_HEARTBEAT_GPIO_PIN            GPIO_Pin_15
-  #define EXTMODULE_HEARTBEAT_EXTI_PortSource     EXTI_PortSourceGPIOD
-  #define EXTMODULE_HEARTBEAT_EXTI_PinSource      GPIO_PinSource15
-  #define EXTMODULE_HEARTBEAT_EXTI_LINE           EXTI_Line15
-  #define EXTMODULE_HEARTBEAT_EXTI_IRQn           EXTI15_10_IRQn
-  #define EXTMODULE_HEARTBEAT_EXTI_IRQHandler     EXTI15_10_IRQHandler
-#endif
+// Unused
+// #if defined(PCBX9LITE)
+//   #define EXTMODULE_HEARTBEAT_RCC_AHB1Periph      RCC_AHB1Periph_GPIOD
+//   #define EXTMODULE_HEARTBEAT_GPIO                GPIOD
+//   #define EXTMODULE_HEARTBEAT_GPIO_PIN            LL_GPIO_PIN_15
+//   #define EXTMODULE_HEARTBEAT_EXTI_PORT           LL_SYSCFG_EXTI_PORTD
+//   #define EXTMODULE_HEARTBEAT_EXTI_SYS_LINE       LL_SYSCFG_EXTI_LINE15
+//   #define EXTMODULE_HEARTBEAT_EXTI_LINE           LL_EXTI_LINE_15
+// #endif
```

Comment:
Why is it unused, and if it is not needed, can it be removed?

Reply:
I'd prefer not to for now, as we don't have any schematics for these, so basically this is the only hardware definition we have.

Reply:
In fact, it seems this has never been used anyway (external module heartbeat does not really exist).